scripts/build_test.py

```diff
@@ -724,26 +724,35 @@ def mock_require_compiled_js_dir_to_be_valid():
                     os.path.exists(os.path.dirname(MOCK_COMPILED_JS_DIR)))
 
     def test_generate_app_yaml(self):
-        app_dev_yaml_file_swap = self.swap(
-            build, 'APP_DEV_YAML_FILE', 'core/tests/data/mock_app_dev.yaml')
-        app_yaml_file_swap = self.swap(
-            build, 'APP_YAML_FILE', 'core/tests/data/mock_app.yaml')
+        mock_dev_yaml_filepath = 'mock_app_dev.yaml'
+        mock_yaml_filepath = 'mock_app.yaml'
+        app_dev_yaml_filepath_swap = self.swap(
+            build, 'APP_DEV_YAML_FILEPATH', mock_dev_yaml_filepath)
+        app_yaml_filepath_swap = self.swap(
+            build, 'APP_YAML_FILEPATH', mock_yaml_filepath)
 
-        self.assertFalse(os.path.isfile('core/tests/data/mock_app.yaml'))
+        temp_file = tempfile.NamedTemporaryFile()
+        temp_file.name = mock_dev_yaml_filepath
+        with open(mock_dev_yaml_filepath, 'w') as tmp:
+            tmp.write('Some content in mock_app_dev.yaml')
+
+        self.assertFalse(os.path.isfile(mock_yaml_filepath))
 
-        with app_dev_yaml_file_swap, app_yaml_file_swap:
+        with app_dev_yaml_filepath_swap, app_yaml_filepath_swap:
             build.generate_app_yaml()
 
-        self.assertTrue(os.path.isfile('core/tests/data/mock_app.yaml'))
+        self.assertTrue(os.path.isfile(mock_yaml_filepath))
 
-        with open('core/tests/data/mock_app.yaml', 'r') as yaml_file:
+        with open(mock_yaml_filepath, 'r') as yaml_file:
             content = yaml_file.read()
 
         self.assertEqual(
             content,
             '# THIS FILE IS AUTOGENERATED, DO NOT MODIFY\n'
-            'Some content in mock_app_dev.yaml.\n')
-        build.safe_delete_file('core/tests/data/mock_app.yaml')
+            'Some content in mock_app_dev.yaml')
+
+        build.safe_delete_file(mock_yaml_filepath)
```

Comment:
Why do you need this? The whole point of what I suggested in the previous review cycle was to not need to create and delete files in the filesystem, so I was wondering if you are able to use a tempfile for mock_yaml_filepath as well?

Reply:
Done